contrib/move-string-resource.kts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * USAGE:
+ * Cuts string resources with the name arg3 from the module from arg1
+ * and pastes them to the string resources from the module from arg2.
+ *
+ * Example usage:
+ * kotlinc -script move-string-resource.kts ../app ../core history_from_current_book
+ *
+ * This will cut all strings with the name history_from_current_book from
+ * all strings.xml files within directories with a prefix of values*
+ * in ../app/src/main/res/ and paste them into
+ * their corresponding files within ../core/src/main/res/values*.
+ *
+ */
+import java.io.File
+import kotlin.system.exitProcess
+
+
+if (args.size < 3) {
+  print_correct_usage_and_exit()
+}
+
+var path_to_values = "/src/main/res/"
+var english_values_dirname = "values"
+var source = args[0]
+var destination = args[1]
+var key = "name=\"" + args[2] + "\""
```

Comment:
kotlin has smart interpolation so you can access variables inside a string using `$` eg "This is the values of $myVariable" or "This is the values of ${myVariable.propertyOfVariable}"

Reply:
oh also you can use a raw string here to avoid escaping because you have `"`s
eg 
```
"""name="${args[2]}""""
```
Or the longer
```
"""
name="${args[2]}"
""".trimIdent()
```

Reply:
@macgills Do you think `File("$destination$path_to_values")` is better style than `File(destination + path_to_values)` ?

Reply:
`File(destination, resDirectory)` is probably for the best
![image](https://user-images.githubusercontent.com/3358282/75260725-f3eee800-57e1-11ea-8ad2-62a48ca14439.png)
File has a lot of constructors to pick from and just String by itself is probably the least helpful

Reply:
oh cool! That is definitely cleaner. Thanks (y)

Reply:
I also might view `pathToValues`/`resDirectory` as not actually useful.

If you just don't set a depth on your filewalk given the module `app` you can return all files called string.xml, if the file contains the string we want create a new file with the path of that file put do a replace on `/app/` with `/core/` and now you are in the right directory. I'd probably have to implement it to see if I like it

Reply:
Hmm, seems like an efficient and clean solution.

Reply:
@macgills Changed it to iterate all files with name "strings.xml" and I think it is an improvement!

```
sourceDir.cutAndPasteStringResourcesToDestination(key, source, destination)

fun File.cutAndPasteStringResourcesToDestination(key: String, source: String, destination: String) {
  this.walk().filter { it.name.equals("strings.xml") }.forEach { resourceFile ->
    var cutLine: String = cutLineFromResourceFile(resourceFile, key)
    if (cutLine.equals("")) {
      // continue for each loop
      return@forEach
    }
    val destinationFile = openOrCreateStringsFile(resourceFile, source, destination)
    pasteLineToDestination(destinationFile, cutLine)
  }
}
```

[See this gist for full source ](https://gist.github.com/Frans-Lukas/d86d867305730cbe349645b829399ee3)